File: app/src/main/java/co/tpcreative/supersafe/model/SyncItemModel.java
package co.tpcreative.supersafe.model;
import java.io.Serializable;

public class SyncItemModel implements Serializable {
    public int id;
    public String originalName;
    public String thumbnailName;
    public boolean isSyncCloud;
    public boolean isSyncOwnServer;
    public String global_original_id;
    public String global_thumbnail_id;
    public String categories_local_id;
    public String categories_id;
    public int formatType;
    public String thumbnailPath;
    public String originalPath;
    public String mimeType;
    public String fileExtension;
    public int statusAction;
    public String items_id;
    public int degrees;
    public boolean thumbnailSync;
    public boolean originalSync;
    public int fileType;
    public String title;
    public String size;
    public boolean isDeleteLocal;
    public boolean isDeleteGlobal;
    public int statusProgress;
    public int deleteAction;
    public boolean isFakePin;
    public boolean isSaver;
    public boolean isExport;
    public boolean isWaitingForExporting;
    public int custom_items;
    public boolean isUpdate;


    /*Added more condition fields*/
    public String global_id;
    public boolean isOriginalGlobalId;

    public SyncItemModel(ItemModel itemModel,boolean isOriginalGlobalId){
        this.id = itemModel.id;
        this.originalName = itemModel.originalName;
        this.thumbnailName = itemModel.thumbnailName;
        this.isSyncCloud = itemModel.isSyncCloud;
        this.isSyncOwnServer = itemModel.isSyncOwnServer;
        this.global_original_id = itemModel.global_original_id;
        this.global_thumbnail_id  = itemModel.global_thumbnail_id;
        this.categories_local_id = itemModel.categories_local_id;
        this.categories_id = itemModel.categories_id;
        this.formatType = itemModel.formatType;
        this.thumbnailPath = itemModel.thumbnailPath;
        this.originalPath = itemModel.originalPath;
        this.mimeType = itemModel.mimeType;
        this.fileExtension = itemModel.fileExtension;
        this.statusAction = itemModel.statusAction;
        this.items_id = itemModel.items_id;
        this.degrees = itemModel.degrees;
        this.thumbnailSync = itemModel.thumbnailSync;
        this.originalSync = itemModel.originalSync;
        this.fileType = itemModel.fileType;
        this.title = itemModel.title;
        this.size = itemModel.size;
        this.isDeleteLocal = itemModel.isDeleteLocal;
        this.isDeleteGlobal = itemModel.isDeleteGlobal;
        this.statusProgress = itemModel.statusProgress;
        this.deleteAction = itemModel.deleteAction;
        this.isFakePin = itemModel.isFakePin;
        this.isSaver = itemModel.isSaver;
        this.isExport = itemModel.isExport;
        this.isWaitingForExporting = itemModel.isWaitingForExporting;
        this.custom_items = itemModel.custom_items;
        this.isUpdate = itemModel.isUpdate;

        /*Added more condition fields*/
        if (isOriginalGlobalId){
            global_id =  itemModel.global_original_id;
        }else{
            global_id =  itemModel.global_thumbnail_id;
        }
        this.isOriginalGlobalId = isOriginalGlobalId;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/model/SyncItemModel.kt
package co.tpcreative.supersafe.model
import co.tpcreative.supersafe.model.ItemModel
import java.io.Serializable

class SyncItemModel(itemModel: ItemModel, isOriginalGlobalId: Boolean) : Serializable {
    var id: Int
    var originalName: String?
    var thumbnailName: String?
    var isSyncCloud: Boolean
    var isSyncOwnServer: Boolean
    var global_original_id: String?
    var global_thumbnail_id: String?
    var categories_local_id: String?
    var categories_id: String?
    var formatType: Int
    var thumbnailPath: String?
    var originalPath: String?
    var mimeType: String?
    var fileExtension: String?
    var statusAction: Int
    var items_id: String?
    var degrees: Int
    var thumbnailSync: Boolean
    var originalSync: Boolean
    var fileType: Int
    var title: String?
    var size: String?
    var isDeleteLocal: Boolean
    var isDeleteGlobal: Boolean
    var statusProgress: Int
    var deleteAction: Int
    var isFakePin: Boolean
    var isSaver: Boolean
    var isExport: Boolean
    var isWaitingForExporting: Boolean
    var custom_items: Int
    var isUpdate: Boolean

    /*Added more condition fields*/
    var global_id: String? = null
    var isOriginalGlobalId: Boolean

    init {
        id = itemModel.id
        originalName = itemModel.originalName
        thumbnailName = itemModel.thumbnailName
        isSyncCloud = itemModel.isSyncCloud
        isSyncOwnServer = itemModel.isSyncOwnServer
        global_original_id = itemModel.global_original_id
        global_thumbnail_id = itemModel.global_thumbnail_id
        categories_local_id = itemModel.categories_local_id
        categories_id = itemModel.categories_id
        formatType = itemModel.formatType
        thumbnailPath = itemModel.thumbnailPath
        originalPath = itemModel.originalPath
        mimeType = itemModel.mimeType
        fileExtension = itemModel.fileExtension
        statusAction = itemModel.statusAction
        items_id = itemModel.items_id
        degrees = itemModel.degrees
        thumbnailSync = itemModel.thumbnailSync
        originalSync = itemModel.originalSync
        fileType = itemModel.fileType
        title = itemModel.title
        size = itemModel.size
        isDeleteLocal = itemModel.isDeleteLocal
        isDeleteGlobal = itemModel.isDeleteGlobal
        statusProgress = itemModel.statusProgress
        deleteAction = itemModel.deleteAction
        isFakePin = itemModel.isFakePin
        isSaver = itemModel.isSaver
        isExport = itemModel.isExport
        isWaitingForExporting = itemModel.isWaitingForExporting
        custom_items = itemModel.custom_items
        isUpdate = itemModel.isUpdate

        /*Added more condition fields*/global_id = if (isOriginalGlobalId) {
            itemModel.global_original_id
        } else {
            itemModel.global_thumbnail_id
        }
        this.isOriginalGlobalId = isOriginalGlobalId
    }
}